sale.py

```diff
@@ -14,3 +14,20 @@ def on_change_with_currency(self, name=None):
             return self.price_list.currency.id
         elif self.company:
             return self.company.currency.id
+
+
+class Line(metaclass=PoolMeta):
+    __name__ = 'sale.line'
+
+    @fields.depends('sale', '_parent_sale.price_list')
+    def _get_context_sale_price(self):
+        context = super()._get_context_sale_price()
+        if self.sale:
+            # sale_price_list module add price_list in the context
+            # In case price_list has currency, set currency from the context to None
+            # This will make that the core module will not try to convert the price
+            # from the price list from company currency to price list currency.
+            if getattr(self.sale, 'price_list', None):
+                if self.sale.price_list.currency:
```

Comment:
Here we should check if sale and price_list currencies differ. If they're not the same, then we either need to convert from one to the other after obtaining the value or raise a UserError telling the user that it is not supported.